packages/playwright/src/migrations/update-18-1-0/remove-baseUrl-from-project-json.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  formatFiles,
+  getProjects,
+  joinPathFragments,
+  type ProjectConfiguration,
+  type Tree,
+  updateProjectConfiguration,
+} from '@nx/devkit';
+
+export default async function (tree: Tree) {
+  const playwrightProjects: ProjectConfiguration[] = [];
+  const projects = getProjects(tree);
+  projects.forEach((project) => {
+    // Check if the project contains playwright config
+    const configPath = joinPathFragments(project.root, 'playwright.config.ts');
+    if (tree.exists(configPath)) {
+      playwrightProjects.push(project);
+    }
+  });
```

Comment:
We shouldn't need to filter projects by the existence of a `playwright.config.ts`. That can leave other valid config file names out. A better and more accurate discriminator would be the executor.

Here we should use the `forEachExecutorOptions` utility instead. Also, we need to account for targetDefaults in `nx.json`.